File: recursion/python/two_branch_recursion.py
# Example case fibonacci number
# f(n) = f(n - 1) + f(n - 2)
# Where f(1) = 1 and f(0) = 0 -> as base case

# O(2^n) time complexity (upper bound)
def fib(n):
  if n <= 1:
    return n
  return fib(n - 1) + fib(n - 2)

File: recursion/cpp/two_branch_recursion.cpp
#include <iostream>

int Fibonacci(int number)
{
	if (number <= 1)
	{
		return number;
	}

	return Fibonacci(number - 1) + Fibonacci(number - 2);
}

int main()
{
	int result = Fibonacci(5);
	std::cout << result;
}